GRIDEDIT-631 restructure repository structure

ScottWillcox opened the pull request.

andreasbuykx reviewed (approved):
I added comments on the individual commits but I'm not sure whether you can see them easily, so here's the summary:

1. there's a few MeshKernelNETCore references left to replace 
2. the commit for moving stuff to root also changed whitespace (presumably line endings?). It would be nice if we would have an automatic cleanup of those on commit or merge.

ScottWillcox commented:
> I added comments on the individual commits but I'm not sure whether you can see them easily, so here's the summary:
> 
>     1. there's a few MeshKernelNETCore references left to replace
> 
>     2. the commit for moving stuff to root also changed whitespace (presumably line endings?). It would be nice if we would have an automatic cleanup of those on commit or merge.

1. I fixed two 'Core' references in the Solution file. Could not find any others.
2. GRIDEDIT-688 should fix some of this. This is running the SharedResharper Configuration on the codebase.
